pkg/ddevapp/hostname_mgt.go

```diff
@@ -110,13 +110,16 @@ func (app *DdevApp) AddHostsEntriesIfNeeded() error {
 			util.Warning("Unable to open hosts file: %v", err)
 			continue
 		}
-		util.Warning("The hostname %s is not currently resolvable, trying to add it to the hosts file", name)
-
-		out, err := escalateToAddHostEntry(name, dockerIP)
-		if err != nil {
-			return err
+		if strings.Contains(name, `*`) {
```

Comment:
We can use this check to cover all possible scenarios (and not run `sudo`):

https://github.com/ddev/ddev/blob/18e53dcc49440f621596387b40d1b942333e1587/vendor/github.com/goodhosts/hostsfile/hosts.go#L187-L189

This will require an update to `go.mod` (move the dependency from the indirect section).

Reply:
I like it.

Reply:
Currently we have used ValidateProjectName for this, https://github.com/ddev/ddev/blob/1daf7841659c3a020d044bb49d73ffa7fe9f6a7d/pkg/ddevapp/config.go#L433-L439